### PR TITLE
feat: return of custom provider

### PR DESCRIPTION
**Pull request by vivalaakam**
## Proposed Changes

- return of custom provider

## Checklist

- [x] I have read the rules of [**contribution**](https://github.com/haqq-network/haqq-wallet/blob/main/docs/contribution.md#important-steps).

### CODE REVIEW

**Review by iGroza — APPROVED** [review]

**Review by devkudasov — APPROVED** [review]

File: e2e/6_sss.test.ts [hunk 1]
```diff
@@ -0,0 +1,220 @@
+import {expect as jestExpect} from '@jest/globals';
+import {by, device, element, expect, waitFor} from 'detox';
+
+import {getTimeStamp} from './helpers/getTimeStamp';
+import {PIN} from './test-variables';
+
+describe('SSS Wallet', () => {
+  let uid = '';
+  let mnemonic = '';
+  let isAndroid = device.getPlatform() === 'android';
```
Comment (ragozin-nikita) [inline e2e/6_sss.test.ts:10]:
```suggestion
  const isAndroid = device.getPlatform() === 'android';
```

File: e2e/6_sss.test.ts [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+
+  beforeAll(() => {
+    if (!uid) {
+      uid = `test${getTimeStamp()}@haqq`;
+    }
+    isAndroid = device.getPlatform() === 'android';
```
Comment (ragozin-nikita) [inline e2e/6_sss.test.ts:16]:
```suggestion
```

File: e2e/6_sss.test.ts [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+
+        if (text) {
+          mnemonic_words.push(text);
+        }
+      } catch (e) {}
+    }
+
+    mnemonic = mnemonic_words.join(' ');
+
+    // await getCoins(mnemonic);
```
Comment (ragozin-nikita) [inline e2e/6_sss.test.ts:114]: Unused code

File: index.js [hunk 4]
```diff
@@ -24,23 +28,20 @@ if (!global.BigInt) {
...
-import {DEBUG_VARS} from '@app/debug-vars';
-import { Feature, isFeatureEnabled } from '@app/helpers/is-feature-enabled';
-import { enableScreens,enableFreeze } from 'react-native-screens';
-
 enableScreens();
 enableFreeze(true);
 
 LogBox.ignoreLogs(["The 'navigation' object hasn't been initialized"]);
+LogBox.ignoreLogs(['Warning: ...']);
+LogBox.ignoreAllLogs();
```
Comment (ragozin-nikita) [inline index.js:36]:
```suggestion
```

File: package.json [hunk 5]
```diff
@@ -15,8 +15,8 @@
     "ts-coverage": "typescript-coverage-report",
     "link-assets": "npx react-native-asset",
     "logcat": "adb logcat -v color -v time --pid $(adb shell ps | grep com.haqq.wallet  | tr -s ' ' | cut -d' ' -f2)",
-    "detox:ios:build": "detox build --configuration ios.sim.release",
-    "detox:ios:test": "detox test --configuration ios.sim.release --loglevel warn --cleanup --record-videos failing --take-screenshots failing --retries 2",
+    "detox:ios:build": "detox build --configuration ios.sim.debug",
```
Comment (ragozin-nikita) [inline package.json:18]:
```suggestion
    "detox:ios:build": "detox build --configuration ios.sim.release",
```

File: package.json [hunk 6]
```diff
@@ -15,8 +15,8 @@
...
+    "detox:ios:test": "detox test --configuration ios.sim.debug --loglevel warn --cleanup --record-videos failing --take-screenshots failing --retries 2",
```
Comment (ragozin-nikita) [inline package.json:19]:
```suggestion
    "detox:ios:test": "detox test --configuration ios.sim.release --loglevel warn --cleanup --record-videos failing --take-screenshots failing --retries 2",
```

File: src/components/browser-privacy-details.tsx [hunk 7]
```diff
@@ -15,6 +15,7 @@ import {MenuNavigationButton, Spacer, Text} from './ui';
 
 export interface BrowserPrivacyDetailsProps {
   permissions: PermissionMap;
+
```
Comment (ragozin-nikita) [inline src/components/browser-privacy-details.tsx:18]:
```suggestion
```

**Review by ragozin-nikita — CHANGES_REQUESTED** [review]

File: e2e/6_sss.test.ts (hunk 3 above)
Comment (vivalaakam) [reply e2e/6_sss.test.ts:111]: used later

File: index.js [hunk 8]
```diff
@@ -24,23 +28,20 @@ if (!global.BigInt) {
...
-import {IS_IOS} from '@app/variables/common';
-import {DEBUG_VARS} from '@app/debug-vars';
-import { Feature, isFeatureEnabled } from '@app/helpers/is-feature-enabled';
-import { enableScreens,enableFreeze } from 'react-native-screens';
-
 enableScreens();
 enableFreeze(true);
 
 LogBox.ignoreLogs(["The 'navigation' object hasn't been initialized"]);
+// LogBox.ignoreLogs(['Warning: ...']);
```
Comment (ragozin-nikita) [inline index.js:35]: Unused code

File: src/components/nft-collection-details.tsx [hunk 9]
```diff
@@ -15,6 +15,7 @@ import {TrimmedText} from './ui/trimmed-text';
 
 export interface NftCollectionDetailsProps {
   item: NftCollection;
+
```
Comment (ragozin-nikita) [inline src/components/nft-collection-details.tsx:18]:
```suggestion
```

File: src/components/signin-networks/signin-networks.tsx [hunk 10]
```diff
@@ -18,15 +16,37 @@ import {SssProviders} from '@app/services/provider-sss';
...
+  onSkip,
+  isAppleSupported,
+  isGoogleSupported,
+  isCustomSupported,
+}: SssNetworksProps) => {
   const [isApple, setIsApple] = useState(false);
   const [isGoogle, setIsGoogle] = useState(false);
+  const [isCustom, setIsCustom] = useState(false);
+  const isLoading = useMemo(
+    () => isApple || isGoogle || isCustom,
```
Comment (ragozin-nikita) [inline src/components/signin-networks/signin-networks.tsx:37]: Seems like this is excessive memoization

File: src/services/provider-sss.ts [hunk 11]
```diff
@@ -5,13 +5,55 @@ import {jsonrpcRequest} from '@haqq/shared-react-native';
...
+      email,
+    }),
+  });
+
+  const authState = await getHttpResponse(token);
+
+  const authInfo = parseJwt(authState.idToken);
+  const verifier = RemoteConfig.get('sss_custom');
+
+  if (!verifier) {
```
Comment (ragozin-nikita) [inline src/services/provider-sss.ts:52]: Maybe you should check this at the beginning of this function?

File: src/variables/sss.ts [hunk 12]
```diff
@@ -3,4 +3,5 @@ import {SssProviders} from '@app/services/provider-sss';
 export const ProviderNameMap = {
   [SssProviders.apple]: 'iCloud',
   [SssProviders.google]: '\nGoogle Drive',
+  [SssProviders.custom]: 'iCloud or Google Drive',
```
Comment (ragozin-nikita) [inline src/variables/sss.ts:6]: I think we should place newline at the beginning and add small adjustment to the name
```suggestion
  [SssProviders.custom]: '\nHAQQ Provider',
```

**Review by ragozin-nikita — CHANGES_REQUESTED** [review]

File: src/types/url-parse.d.ts [hunk 13]
```diff
@@ -17,10 +17,15 @@ declare module 'url-parse' {
     constructor(address: string, parser?: boolean | Function);
 
     extractProtocol(url: string): any;
+
```
Comment (ragozin-nikita) [inline src/types/url-parse.d.ts:20]:
```suggestion

```

File: src/services/logger.ts [hunk 14]
```diff
@@ -37,6 +33,12 @@ export class LoggerService {
     return `${this.emodji}`;
   }
 
+  private _emodji?: string;
```
Comment (ragozin-nikita) [inline src/services/logger.ts:36]: Why this was moved?

**Review by ragozin-nikita — CHANGES_REQUESTED** [review]

**Review by ragozin-nikita — APPROVED** [review]

**Pull request merged**